tools/toolchains/arm.py

```diff
@@ -342,9 +342,9 @@ def __init__(self, target, notify=None, macros=None,
                  build_profile=None, build_dir=None):
         ARM.__init__(self, target, notify, macros, build_dir=build_dir,
                      build_profile=build_profile)
-        if not set(("ARM", "uARM")).intersection(set(target.supported_toolchains)):
-            raise NotSupportedException("ARM/uARM compiler support is required for ARM build")
-
+        #check only for ARMC5 because ARM_STD means using ARMC5, and thus supported_toolchains must include ARMC5
+        if "ARMC5" not in target.supported_toolchains:
+            raise NotSupportedException("ARM compiler 5 support is required for ARM build")
```

Comment:
This undermines much of your work above to be backwards compatible with prior version of Mbed OS. This will raise an exception in the case that we don't have `ARMC5` in the supported toolchain list, which is always true for all prior versions of Mbed OS.